charts/fluent-bit/templates/ingress.yaml

```diff
@@ -0,0 +1,62 @@
+{{- $ingressApiIsStable := eq (include "fluent-bit.ingress.isStable" .) "true" -}}
+{{- $ingressSupportsIngressClassName := eq (include "fluent-bit.ingress.supportsIngressClassName" .) "true" -}}
+{{- $ingressSupportsPathType := eq (include "fluent-bit.ingress.supportsPathType" .) "true" -}}
+{{- $fullName := include "fluent-bit.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+
+{{- if .Values.ingress.enabled -}}
```

Comment:
Do we want to gate this so it only works when deployment mode is being used?

Reply:
For my use-case, deployment mode is enough. But will works fine with daemonset as well, depends on fluent-bit config, so it's your choice.

Reply:
I'm saying that ingress should only be enabled in deployment mode, I can't see any use case for needing it in daemonset mode. Also the service should be removed when running in daemonset mode as we should be using a `PodMonitor` and not a `ServiceMonitor`, we don't need the extra overhead of a service.

Reply:
I would not remove service in daemonset mode. Imagine you have http input, you want to POST to this input via cluster local svc endpoint, but preferably to nearest fluent-bit pod ( same node ) using svc topology keys...

Reply:
That is exactly why the service should be removed. If you want aggregation functionality run an aggregator, collectors are specifically for that, collecting logs directly from a node without the additional overhead required to take arbitrary inputs.

Reply:
Ok, but why not wait with this decisions till there will be aggregator chart version?

Reply:
I'm happy to approve ingress for deployment mode, but not for daemonset mode.

Reply:
Ok, done.